exec/java-exec/src/main/java/org/apache/drill/exec/server/options/TypeValidators.java

```diff
@@ -179,6 +181,7 @@ public void validate(final OptionValue v) {
 
     public TypeValidator(final String name, final Kind kind, final OptionValue defValue) {
       super(name);
+      checkArgument(defValue.type == OptionType.SYSTEM, "Default value must be SYSTEM type.");
```

Comment:
There are three other checkArguments in this changeset. However, I think they also refer to dev mistakes instead of user errors.  Probably fine as is.